### PR TITLE
feat:[#31] update dependent entities logic

### LINKED ISSUE #31: Provider-Service Connection Logic
- [ ] Establish the logic for connecting providers and services. When a provider is unpublished, all associated services should also be unpublished. Schedule a meeting to finalize this logic with double validation. Ala to design this process.
https://docs.cyfronet.pl/display/FID/Object+dependencies+and+logic

### PR DESCRIPTION
**Pull request by goreck888**
Modify suspend/unpublish/delete actions
Now actions listed above on catalogue/provider
force the same action on dependent providers/services

Closes #31

### CODE REVIEW

**Comment by Marcelinna** [comment]
1. Create Catalogue - Catalogue has **published** status
2. Create a Provider linked to the Catalogue - Provider has **published** status
3. Change Catalogue status from **published** to **suspended**. Provider has status **suspended** 
4. Change catalogue status to **published**. Provider has status **suspended**. 
Provider should has published status

**Comment by Marcelinna** [comment]
image
1. Click suspend on catalogue
image
2. Click unpublish on catalogue
3. See problem - catalogue is still suspended
image

**Comment by Marcelinna** [comment]
1. User has catalogue with Published status
2. User has Provider connected to catalogue with Suspended status
3. Click Unpublish on Catalogue
4. User see notification _Are you sure you want to unpublish this catalogue? It will unpublish all dependent providers and services._, in this case it is no true, because Provider is suspended. @goreck888 @AlicjaSwierad we need better message for the user
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

1. The Delete button on the catalogue doesn't work

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
1. It is possible to create a service for the deleted provider

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
the user should be able to assign published/unpublished/suspended catalogs to the provider,
as in the case of services

File: test/system/cypress/e2e/regression_tests/backoffice/providers/providers.spec.ts [hunk 1]
```diff
@@ -113,25 +113,25 @@ describe("Providers", () => {
     cy.contains("div.alert-success", message.successDeletionMessage).should("be.visible");
   });
 
-  it("shouldn't delete provider with services with draft status", () => {
+  it.skip("shouldn't delete provider with services with draft status", () => {
```
Comment (Michal-Kolomanski) [inline test/system/cypress/e2e/regression_tests/backoffice/providers/providers.spec.ts:116]: Can you add a skip reason here? "According to new business logic, that requirement is no longer valid" 
Or just delete those tests

File: app/helpers/forms_helper.rb [hunk 2]
```diff
@@ -69,4 +69,14 @@ def render_public_contact(public_contact_form, object)
            object: object,
            provider_form: object.is_a?(Provider)
   end
+
+  def action_prompt(object, action)
+    entities = object == "catalogue" ? "providers and services" : "services"
```
Comment (Michal-Kolomanski) [inline app/helpers/forms_helper.rb:74]: Maybe we should also add offers here because they are also deleted

**Review by Michal-Kolomanski — APPROVED** [review]
Code looks really clean & natty 👏

File: app/helpers/forms_helper.rb (hunk 2 above)
Comment (goreck888) [reply app/helpers/forms_helper.rb:74]: Sure

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
image
image

Please change the message similar to the Delete button on the Provider page
image

File: app/helpers/forms_helper.rb (hunk 2 above)
Comment (Michal-Kolomanski) [reply app/helpers/forms_helper.rb:74]: @Marcelinna that can be discussed and agreed upon, but its a separate talk I believe. For now, offers are just deleted permanently. What can be done in this PR is to inform users that offers will also be deleted

Comment (goreck888) [reply app/helpers/forms_helper.rb:74]: Now it's done.

Comment (Michal-Kolomanski) [reply app/helpers/forms_helper.rb]: add bundles

**Pull request merged**